plugin/rewrite/reverter.go

```diff
@@ -21,23 +22,69 @@ type ResponseRule struct {
 // dig will complain with ';; Question section mismatch: got example.org/HINFO/IN'
 type ResponseReverter struct {
 	dns.ResponseWriter
-	originalQuestion dns.Question
-	ResponseRewrite  bool
-	ResponseRules    []ResponseRule
+	request           *dns.Msg
```

Comment:
not only do you put the original message in there; you're duplicating original question and original opt header as well. 
dont understand

Reply:
That is the whole point of this PR: (see description of the PR).

We need that `rewrite` return the `OPT Header` part of the `request` as it was before any modification by Rewrite. (because SizeAndDo will response to initial client according to that Header).

for that we need to keep a copy of the original opt header.

the `request` is only a pointer to the request that is flowing up to the pugin resolver. It can be changed by any of the plugin in the handler list.


About copy of `QUESTION` I guess that is a similar problem. But was there before this PR.